test/Microsoft.IdentityModel.Tests/ReferenceXml.cs

```diff
@@ -39,93 +39,6 @@ public class ReferenceXml
     {
         #region EnvelopedSignatureReader / Writer
 
-        public static string Saml2Token_TwoSignatures
```

Comment:
I removed it because the current check was in the signature verification layer and that code is now removed.

If you still want an error for duplicate signatures in a SAML2 token / assertion that is IMHO rather a schema validation issue and should be handled by code that knows it is a SAML2 token that is read. The `EnvelopedSignatureReader` could be used to read other data than an Assertion. In face, that's the point with this PR to be able to use it to verify signatures over an entire SAML2 response.

I can look into adding back such a test, but move the check to the `Saml2Serializer`.

Reply:
@brentschmaltz This check is what I need feedback on to know how to proceed. I think that a viable option is to just keep it as it is in the PR right now. There are cases where multiple signatures are allowed even inside an assertion. And the code still properly validates that the data is signed by a trusted authority.